substrate/frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -245,6 +245,21 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 			}
 		});
 
+	let deprecation = if let Some(call) = def.call.as_ref() {
```

Comment:
nit: Maybe we can remove the if statement here.

Did not check this for compiling, similar to:
```rust
let attributes = def.call.as_ref().map(|call| call.attrs.as_ref()).unwrap_or(&[]);
// then we can call get_deprecated_enum once

crate::deprecation::get_deprecation_enum(
			&quote::quote! {#frame_support},
			attributes,
			methods.iter().map(|item| (item.call_index as u8, item.attrs.as_ref())),
		)
```

Reply:
done as suggested